webapp/src/components/ProgramActivityDialog.tsx

```diff
@@ -1,85 +1,212 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { decodeHTML } from 'entities';
+import CancelIcon from '@mui/icons-material/Cancel';
+import CloseIcon from '@mui/icons-material/Close';
+import TaskAltIcon from '@mui/icons-material/TaskAlt';
+import LoadingButton from '@mui/lab/LoadingButton';
 import {
-  Button,
+  Alert,
   Dialog,
   DialogActions,
   DialogContent,
   DialogContentText,
   DialogTitle,
   Divider,
+  FormGroup,
+  Snackbar,
   Table,
   TableBody,
   TableCell,
   TableContainer,
   TableRow,
 } from '@mui/material';
-
+import IconButton from '@mui/material/IconButton';
 import { formatDate, formatTime } from '../helpers/dateTime';
 import { CalendarEvent } from '../types/api';
-
 interface ProgramActivityDialogProps {
   show: boolean;
   contents: CalendarEvent;
   handleClose: () => void;
 }
+const sendAPIGetRequest = (
+  path: string,
+  activityType: string,
+  setCompleted: React.Dispatch<React.SetStateAction<boolean | null>>,
+  setError: React.Dispatch<React.SetStateAction<boolean | null>>,
+  setIsErrorShown: React.Dispatch<React.SetStateAction<boolean>>,
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>
+) => {
+  setIsLoading(true);
+  if (activityType !== 'assignment') return;
+  return fetch(`${process.env.REACT_APP_API_ORIGIN}${path}`, {
+    method: 'GET',
+    credentials: 'include',
+  })
+    .then(res => res.json())
+    .then(
+      async ({ data }) => {
+        if (data) {
+          setIsLoading(false);
+          setError(false);
+          setCompleted(data.completed);
+        }
+      },
+      error => {
+        setIsLoading(false);
+        setError(true);
+        setIsErrorShown(true);
+      }
+    );
+};
+
+const sendAPIPutRequest = (
+  path: string,
+  body: Object,
+  setCompleted: React.Dispatch<React.SetStateAction<boolean | null>>,
+  setIsUpdateSuccess: React.Dispatch<React.SetStateAction<boolean>>,
+  setIsMessageVisible: React.Dispatch<React.SetStateAction<boolean>>,
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>
+) => {
+  setIsLoading(true);
+  if ('completed' in body && body.completed === null) {
```

Comment:
wondering if only me got this warning 'Property 'completed' does not exist on type 'Object'.'

<img width="770" alt="image" src="https://user-images.githubusercontent.com/83237024/207671193-b5f97da0-f885-435e-a5c6-53ae824afd40.png">

Reply:
It is showing on my machine too.

Reply:
it's not appearing on my end